e2e_deployment_files/edgehub_restart_deployment.template.json

```diff
@@ -0,0 +1,217 @@
+            },
+            "env": {
+              "CollectMetrics": {
+                "value": "true"
+              },
+              "experimentalfeatures__enabled": {
+                  "value": "true"
+              },
+              "experimentalfeatures__enableMetrics": {
+                  "value": "true"
+              }
+            },
+            "status": "running",
+            "restartPolicy": "always"
+          }
+        },
+        "modules": {
+          "directMethodReceiver1": {
+            "version": "1.0",
+            "type": "docker",
+            "status": "running",
+            "restartPolicy": "always",
+            "env": {
+              "ClientTransportType": {
+                "value": "Amqp"
+              },
+              "ReportingEndpointUrl": {
+                "value": "http://testResultCoordinator:5001"
+              },
+              "trackingId": {
+                "value": "<TrackingId>"
+              }
+            },
+            "settings": {
+              "image": "<Container_Registry>/microsoft/azureiotedge-direct-method-receiver:<Build.BuildNumber>-linux-<Architecture>",
+              "createOptions": ""
+            }
+          },
+          "edgeHubRestartTester1": {
+            "version": "1.0",
+            "type": "docker",
+            "status": "running",
+            "restartPolicy": "always",
+            "env": {
+              "IOT_HUB_CONNECTION_STRING": {
+                "value": "<IoTHubConnectionString>"
+              },
+              "directMethodEnabled": {
+                "value": "true"
+              },
+              "directMethodTargetModuleId": {
+                "value": "directMethodReceiver1"
+              },
+              "messageEnabled": {
+                "value": "true"
+              },
+              "reportingEndpointUrl": {
+                "value": "http://localhost:5001"
+              },
+              "restartPeriod": {
+                "value": "<EdgeHubRestartTest.RestartPeriod>"
+              },
+              "sdkOperationTimeout": {
+                "value": "<EdgeHubRestartTest.SdkOperationTimeout>"
+              },
+              "testDuration": {
+                "value": "<TestDuration>"
+              },
+              "testStartDelay": {
+                "value": "<TestStartDelay>"
+              },
+              "trackingId": {
+                "value": "<TrackingId>"
+              }
+            },
+            "settings": {
+              "image": "<Container_Registry>/microsoft/azureiotedge-edgehub-restart-tester:<Build.BuildNumber>-linux-<Architecture>",
+              "createOptions": "{\"HostConfig\":{\"NetworkMode\":\"host\"},\"NetworkingConfig\":{\"EndpointsConfig\":{\"host\":{}}}}"
+            }
+          },
+          "relayer1": {
```

Comment:
Why do we need the relayer?

Reply:
It is use as a dummy receiver module of the message. This allows us to verify the message actually reaches the destination. The Relayer1 forward the message from our sender to the TRC for the verification against the sender's record that the message was sent and received.

Reply:
Also since this is messaging, the sender module immediately stop throwing exception once the EdgeHub is up (which is the point of connectivity test) -- it doesn't wait for the SDK to re-connect the relayer as a receiver.